### PR TITLE
feat : Adds Github Link , makes code modular , improves  payout UI significantly

### LINKED ISSUE #1057: feature: make user link their github here
Screenshot 2024-08-27 at 5 07 07 PM

Should be a button that says `link your github` that takes user through auth flow to link it.

That way, whenever a bounty is given it can automatically reach the specific users solana address

### PR DESCRIPTION
**Pull request by amanbairagi30**
### PR Fixes:
- 1 Add Github Link
- 2 Makes Code Modular 
- 3 Improves payout UI significantly (it is  responsive as well)

Resolves #1057

image

Below is how it looks like

https://github.com/user-attachments/assets/c357718e-6861-4ffa-acb5-d3227d089548

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

**Comment by amanbairagi30** [comment]
@hkirat @devsargam

**Comment by ItsFlash10** [comment]
Good work dude

**Comment by amanbairagi30** [comment]
> Good work dude

Thanks ser 🚀

File: src/app/api/github/details/route.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { NextResponse } from "next/server";
+
+export async function GET() {
+    const session = await getServerSession(authOptions);
+
+    if (!session) {
+        return NextResponse.json({ message: 'Unauthorised', success: 'false' }, { status: 401 });
+    }
+
+    const githubData = await prisma.gitHubLink.findMany({
```
Comment (hkirat) [inline src/app/api/github/details/route.ts:13]: findOne?

**Comment by amanbairagi30** [comment]
> /bounty $75

Thank you bhaiya

**Pull request merged**